dandi/support/digests.py

```diff
@@ -85,8 +85,13 @@ def __call__(self, fpath: Union[str, Path]) -> Dict[str, str]:
 def get_digest(filepath: Union[str, Path], digest: str = "sha256") -> str:
     if digest == "dandi-etag":
         return cast(str, get_dandietag(filepath).as_str())
-    else:
-        return Digester([digest])(filepath)[digest]
+    elif digest == "zarr-checksum":
+        path = Path(filepath)
+        if path.is_dir():
+            return get_zarr_checksum(path)
+        else:
+            digest = "md5"
```

Comment:
shouldn't this logic just fold into `get_zarr_checksum` since IMHO it leaks implementation detail (of "zarr-checksum") into outside code.

Reply:
The only implementation detail here is the use of MD5.  `get_zarr_checksum()` is only meant to operate on directories.

Reply:
so it remains confusing why `get_digest(digest="zarr-checksum")` can operate on a file while `get_zarr_checksum()` can't. IMHO should be consistent one way or another. Do we need to allow `get_digest(digest="zarr-checksum")` to operate on a file or could raise an error?

Reply:
> Do we need to allow `get_digest(digest="zarr-checksum")` to operate on a file or could raise an error?

I don't really care either way.

Reply:
`get_zarr_checksum()` already raises an error if it's passed a file.  Did you mean `get_digest(path, digest = "zarr-checksum")`?

Reply:
yes. why should it behave differently?

Reply:
Because `get_digest()` is called by `dandi digest`, and users are likely to try to run `dandi digest -d zarr-checksum` on a file.

Reply:
so -- if the checksum is defined for `zarr` then it should also be reported by `get_zarr_checksum`.  If it is not defined -- neither should report it.  If we decide to go "not defined" (keep exception in `get_zarr_checksum`) then `dandi digest -d zarr-checksum` should error out

Reply:
`get_zarr_checksum()` now accepts file paths.